RecoEgamma/EgammaElectronAlgos/interface/TrajSeedMatcher.h

```diff
@@ -156,6 +185,32 @@ class TrajSeedMatcher {
     const std::vector<double> dRZMaxLowEtEtaBins_; 
     const std::vector<double> dRZMaxLowEt_; 
   };
+  class MatchingCutsV2 : public MatchingCuts {
+  public:
+    explicit MatchingCutsV2(const edm::ParameterSet& pset);
+    bool operator()(const SCHitMatch& scHitMatch)const;
+  private:
+    const egPM::Param<SCHitMatch> dPhiMin_;
+    const egPM::Param<SCHitMatch> dPhiMax_;
+    const egPM::Param<SCHitMatch> dRZMin_;
+    const egPM::Param<SCHitMatch> dRZMax_;
+  }; 
+
+  class MatchingCutsV3 : public MatchingCuts {
```

Comment:
I dont feel strongly here. I choose V{1,2,3} simply as I hate naming things (and as you are now no doubt painfully aware, fairly bad at it too). So pixel match selection version 1, 2, 3 and so on. Rather than trying to have descriptive names which might cause me problems down the line ( I can see it going "EtGradSmallVariation". So I can have in my config just "version==1" and then it should hopefully be obvious from the parameters supplied. 

Here are some example configs:
this uses those objects, was an experimental WP
https://sharper.web.cern.ch/sharper/cms/electronSeedsWP7.py

this is the WP proposed for the next menu WP
https://sharper.web.cern.ch/sharper/cms/electronSeedsWP8.py

I'm okay with however you rule, I see pros and cons for both approaches